src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -498,9 +499,9 @@ bool CascadiaSettings::_isPowerShellCoreInstalled(std::filesystem::path& cmdline
 // - A ref of a path that receives the result of PowerShell Core pwsh.exe full path.
 // Return Value:
 // - true iff powershell core (pwsh.exe) is present in the given path
-bool CascadiaSettings::_isPowerShellCoreInstalledInPath(const std::wstring_view programFileEnv, std::filesystem::path& cmdline)
+bool CascadiaSettings::_isPowerShellCoreInstalledInPath(_In_ PCWSTR programFileEnv, std::filesystem::path& cmdline)

```

Comment:
i'd prefer we stick with `wstring_view` even though it doesn't get us anything here but some measure of misguided type safety. If eventually we wanted to know the size, it'd all be right there in the wstring_view for us

Reply:
Will change back.

I made the change because otherwise I would have had to copy the value into a writable `wstring` and call .data()/.c_str() on that instead of just calling .data() on the `wstring_view`. That's because string_view just saves a pointer and length but doesn't check that the pointer is to a "proper C string" (null-terminated string), which the WIL helper demands.

(Technically, the way this private method actually gets called, we know the string view only ever gets a null terminated string. But then I'd have to convince /analyze of this with _Analysis_assume_, and then add a comment, and then...)